assets/app/data/crowbar/steps.factory.js

```diff
@@ -4,72 +4,107 @@
         .module('suseData.crowbar')
         .factory('upgradeStepsFactory', upgradeStepsFactory);
 
-    upgradeStepsFactory.$inject = [];
+    upgradeStepsFactory.$inject = ['$state'];
     /* @ngInject */
-    function upgradeStepsFactory() {
-        var steps = initialSteps(),
-            factory = {
-                getAll: getSteps
-            };
+    function upgradeStepsFactory($state) {
+        var factory = {
+            steps: initialSteps(),
+            activeStep: {},
+            refeshStepsList: refeshStepsList,
+            setCurrentStepCompleted: setCurrentStepCompleted,
+            stepFinished: stepFinished
+        };
 
         return factory;
 
+        function stepFinished() {
+            return factory.activeStep.finished;
+        }
+        function setCurrentStepCompleted() {
+            factory.activeStep.finished = true;
+        }
+
+        function refeshStepsList() {
+            var currentState = $state.current.name,
+                isCompletedStep = true;
```

Comment:
This var was used in the prototype to mark the completed steps in green. See https://github.com/crowbar/crowbar-ui/blob/master/assets/app/widgets/crowbar-upgrade-steps/crowbar-upgrade-steps.directive.jade#L2

It's a miss from the cleanup card and will be removed on Richa's changes